### PR TITLE
Add missing table `id` in DRAGEN modules, require them in linting

### PR DESCRIPTION
**Pull request by vladsavelyev**
Add missing `pconfig` with `namespace` and `id` in DRAGEN submodules. Fixes settings table column visibility configuration, e.g.

```yaml
table_columns_visible:
  dragen-trimmer-metrics-table:
    Total input reads: False
    Unmapped reads pct: False
    Number of duplicate marked reads pct: False
    Total: False
    Total input bases R2: False
```

Fixes https://github.com/ewels/MultiQC/issues/1954

Require `id` to be set in pconfig when `--strict`. When it's not set, auto-generate random `id` earlier to make the config options work.

As some additional refactoring, removed warnings of missing `r` prefixes in regex strings.

TODO: 
- [x] now fix the rest of the linting :)

### CODE REVIEW

File: multiqc/modules/dragen/coverage_metrics.py [hunk 1]
```diff
@@ -901,10 +901,12 @@ def make_own_coverage_sections(coverage_data, coverage_headers, bed_texts):
                     + bed_texts[phenotype]["description"]
                     + "\n\nPress the `Help` button for details."
                 )
+                anchor = "dragen-cov-metrics-own-section-" + re.sub(r"(\s|-|\.|_)+", "-", phenotype)
+                plots[phenotype]["config"]["id"] = anchor + "-table"
```
Comment (ewels) [inline multiqc/modules/dragen/coverage_metrics.py:905]:
```suggestion
                plots[phenotype]["config"]["id"] = f"{anchor}-table"
```

File: multiqc/modules/hicpro/hicpro.py [hunk 2]
```diff
@@ -289,15 +306,24 @@ def hicpro_mapping_chart(self):
...
+                d = dict()
+                if n_global is not None:
+                    d["Full_Alignments_Read"] = n_global
+                if n_local is not None:
+                    d["Trimmed_Alignments_Read"] = n_local
+                if n_mapped is not None and n_total is not None:
+                    try:
+                        d["Failed_To_Align_Read"] = int(n_total) - int(n_mapped)
+                    except ValueError:
+                        log.debug(
```
Comment (ewels) [inline multiqc/modules/hicpro/hicpro.py:322]: This shouldn't happen very often, right?
```suggestion
                        log.warning(
```

File: multiqc/modules/hicpro/hicpro.py [hunk 3]
```diff
@@ -169,7 +186,7 @@ def parse_hicpro_stats(self, f, rsection):
                         self.hicpro_data[s_name][s[0]] = float(s[1])
                     # Otherwise just store the value as is.
                     except ValueError:
-                        self.hicpro_data[s_name][s[0]] = s[1]
+                        log.error(f"Could not parse value for key '{s[0]}' as a number in sample '{s_name}': '{s[1]}'")
```
Comment (ewels) [inline multiqc/modules/hicpro/hicpro.py:189]: Do we want to keep the value even if it's not a number?

File: multiqc/modules/qc3C/qc3C.py [hunk 4]
```diff
@@ -996,7 +996,7 @@ def _none_to(x, y):
             self.add_data_source(f, s_name, section=analysis_mode)
 
         except KeyError as ex:
-            log.error(
+            log.debug(
```
Comment (ewels) [inline multiqc/modules/qc3C/qc3C.py:999]: Yes, this will help the CI log outputs look nicer! 😅

**Review by ewels — APPROVED** [review]
Looks good! Couple of minor questions but nothing big.

File: multiqc/modules/hicpro/hicpro.py (hunk 2 above)
Comment (vladsavelyev) [reply multiqc/modules/hicpro/hicpro.py:322]: Yeah, agree

File: multiqc/modules/hicpro/hicpro.py (hunk 3 above)
Comment (vladsavelyev) [reply multiqc/modules/hicpro/hicpro.py:189]: I was thinking that they are attempted to be parsed as numbers for some calculations further in the code, so there is no reason to keep strings. But perhaps there are other metrics that are not used for calculations and we want to keep, so I'll revert and test this bit

Comment (vladsavelyev) [reply multiqc/modules/hicpro/hicpro.py:191]: Reverted that, and did more debugging in hicpro, there were actually a few issues with it:
- the search patterns didn't account for half of input test data (e.g. `.mmapstat` was incouded by `.mapstat` was missing; `_assplit.stat` was missing, etc)
- there were warning about missing data on plots, just needed to add a check that the required data was found like in other modules
- improved debug info about duplicated key for a sample

The PR is heavy in single module, perhaps should split it up for clarity 🤔

Comment (vladsavelyev) [reply multiqc/modules/hicpro/hicpro.py]: Moving this stuff into a separate PR

**Comment by vladsavelyev** [comment]
@multiqc-bot changelog

**Pull request merged**